### PR TITLE
wip: better rhino > revit brep support

### PR DESCRIPTION
**Pull request by didimitrie**
[CNX-574: Better brep support rhino > revit](https://linear.app/speckle/issue/CNX-574/better-brep-support-rhino-revit)
Pair sdk pr: https://github.com/specklesystems/speckle-sharp-sdk/pull/139

What this PR brings is better brep support in the following workflows: 
- rhino <> rhino
- rhino > revit

Needless to say, the last one is the crucial improvement as it allows us receiving "as close to native" solid elements into Revit. This has been heavily inspired by the work over in the Rhino Inside world. 

The tl;dr of the change: we are now transporting breps (and co) by encoding them in a rhino3dm file, and using the shape importer class from revit's sdk to load them in.

As this PR is a bit fat, I'm happy to do a live review and explain + edit session. Also note, the receive side of revit is currently a bit of a hot mess. 

~~Extraneous: rhino test project fails as the fakes don't contain new used namespaces. my proposoal: since rhinocommon is so nice to us, should we just swap the fakes to it?~~

### CODE REVIEW

**Comment by AlanRynne** [comment]
👀 amazing, i'll have a good look at this tomorrow 🙂

File: Converters/Rhino/Speckle.Converters.RhinoShared/ToHost/Raw/BrepXToHostConverter.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+using Speckle.Converters.Common.Objects;
+
+namespace Speckle.Converters.Rhino.ToHost.Raw;
+
+public class BrepXToHostConverter : ITypedConverter<SOG.BrepX, List<RG.GeometryBase>>
+{
+  public List<RG.GeometryBase> Convert(SOG.BrepX target) => RawEncodingToHost.Convert(target);
+}
+
+public class SubDXToHostConverter : ITypedConverter<SOG.SubDX, List<RG.GeometryBase>>
```
Comment (oguzhankoral) [inline Converters/Rhino/Speckle.Converters.RhinoShared/ToHost/Raw/BrepXToHostConverter.cs:12]: i should have split them into different classes, would be nice if you can, but trivial for now

File: Converters/Rhino/Speckle.Converters.RhinoShared/ToSpeckle/Raw/BrepToSpeckleConverter.cs [hunk 2]
```diff
@@ -43,7 +43,7 @@ IConverterSettingsStore<RhinoConversionSettings> settingsStore
   public SOG.Brep Convert(RG.Brep target)
   {
     var tol = _settingsStore.Current.Document.ModelAbsoluteTolerance;
-    target.Repair(tol);
+    //target.Repair(tol);
```
Comment (oguzhankoral) [inline Converters/Rhino/Speckle.Converters.RhinoShared/ToSpeckle/Raw/BrepToSpeckleConverter.cs:46]: to not forget later, would be nice if you create ticket to have it as setting if it make sense? i don't remember what we agreed on before

**Review by oguzhankoral — CHANGES_REQUESTED** [review]
Minors

File: Connectors/Revit/Speckle.Connectors.RevitShared/Operations/Receive/RevitHostObjectBuilder.cs [hunk 3]
```diff
@@ -107,55 +112,54 @@ CancellationToken cancellationToken
...
-      _logger.LogError(ex, "Failed to create group after receiving elements in Revit");
+      _activityFactory.Start("Painting solids");
+      _transactionManager.StartTransaction(true, "Painting solids");
+      PostBakePaint(conversionResults.postBakePaintTargets);
+      _transactionManager.CommitTransaction();
     }
 
-    using (var _ = _activityFactory.Start("Commit"))
+    // 5 - Create group
     {
```
[the remark this answers is not in the document]
Comment (didimitrie) [reply Connectors/Revit/Speckle.Connectors.RevitShared/Operations/Receive/RevitHostObjectBuilder.cs:149]: purely style chance i tried out to see if it looks "cleaner". empty curlyies define a scope, where in the `using var` gets disposed of at the end. ps: accidentaly re-requested review, it's not ready yet for that

File: Converters/Rhino/Speckle.Converters.RhinoShared/ToSpeckle/Raw/BrepToSpeckleConverter.cs (hunk 2 above)
Comment (didimitrie) [reply Converters/Rhino/Speckle.Converters.RhinoShared/ToSpeckle/Raw/BrepToSpeckleConverter.cs:46]: uncommented and added note on why it's potentially tricky

**Review by oguzhankoral — APPROVED** [review]
